source/s3_client.c

```diff
@@ -1676,8 +1676,9 @@ void aws_s3_client_update_connections_threaded(struct aws_s3_client *client, boo
         uint32_t num_conns_per_vip = s_num_conns_per_vip_meta_request_look_up[request->meta_request->type];
         const uint32_t max_active_connections = aws_s3_client_get_max_active_connections(client, num_conns_per_vip);
 
-        /* If this meta request was finished, then let the meta request know and release the request. */
-        if (aws_s3_meta_request_has_finish_result(request->meta_request)) {
+        /* Unless the request is marked "always send", if this meta request is finished, then finish the request now and
+         * release it. */
+        if (!request->always_send && aws_s3_meta_request_has_finish_result(request->meta_request)) {
```

Comment:
Haven't dived deep into it, can you birefly explain about why it's checking the finish_result instead of `aws_s3_meta_request_is_finished`?

Reply:
has_finish_result = We know that this thing failed or succeeded, but it could still be winding down and have in flight requests.
is_finished = Nothing left for this thing to do at all.

Reply:
Yeah, maybe add it to the commends? Since, the comments saying `if this meta request is finished` is confusing as well.

Reply:
Added it!